File: tests/test_distribution/check.R
#!/usr/bin/env Rscript
require(ggplot2)
require(data.table)

mpfr = read.table("out_mpfr", header=T)
quad = read.table("out_quad", header=T)

mpfr$backend=factor("mpfr")
quad$backend=factor("quad")

data = rbind(mpfr, quad) 
print(summary(data))

print(sd(data[data$backend=="mpfr",]$c))
print(sd(data[data$backend=="quad",]$c))

print(summary(data))

ggplot(data, aes(x=backend, y=c)) + geom_violin() + geom_jitter(alpha=0.5) 

File: tests/test_distribution/check.py
#!/usr/bin/env python

THRESHOLD = 1.9

import sys

for quad, mpfr in zip(file("out_quad"), file("out_mpfr")):
    q = float(quad)
    m = float(mpfr)
    if q-m > THRESHOLD:
        print "ERROR:", q, m
        sys.exit(1)

sys.exit(0)
